doc/examples/segmentation/plot_rolling_ball.py

```diff
@@ -0,0 +1,270 @@
+ax[1].set_title('Correct Background Removal')
+ax[1].axis('off')
+
+fig.tight_layout()
+
+plt.show()
+
+######################################################################
+# Image Datatypes
+# ---------------
+#
+# ``rolling_ball`` can handle datatypes other than `np.uint8`. You can
+# pass them into the function in the same way.
+
+image = data.coins().astype(np.uint16)
+
+background = morphology.rolling_ball(image, radius=70.5)
+plot_result(image, background)
+plt.show()
+
+######################################################################
+# However, you need to be careful if you use floating point images
+# that have been normalized to ``[0, 1]``. In this case the ball will
+# be much larger than the image intensity, which can lead to
+# unexpected results.
+
+image = util.img_as_float(data.coins())
+
+background = morphology.rolling_ball(image, radius=70.5)
+plot_result(image, background)
+plt.show()
+
+######################################################################
+# Because ``radius=70.5`` is much larger than the maximum intensity of
+# the image, the effective kernel size is reduced significantly, i.e.,
+# only a small cap (approximately ``radius=10``) of the ball is rolled
+# around in the image. You can find a reproduction of this strange
+# effect in the ``rolling_ellipsoid`` section below.
+#
+# To get the expected result, you need to use the more flexible
+# ``rolling_ellipsoid`` function. It works just like ``rolling_ball``;
+# however, it gives you more control over the affected area and
+# strength of the algorithm. In particular, it has different
+# parameters for the spatial dimensions and the intensity dimension of
+# the image.
+
+image = util.img_as_float(data.coins())
+
+normalized_radius = 70.5 / 255
+background = morphology.rolling_ellipsoid(
+    image,
+    kernel_shape=(70.5 * 2, 70.5 * 2),
+    intensity_vertex=normalized_radius * 2
+)
+plot_result(image, background)
+plt.show()
+
+######################################################################
+# Rolling Ellipsoid
+# -----------------
+#
+# In ``rolling_ellipsoid`` you are specifying an ellipsoid instead of
+# a ball/sphere - sidenote: a ball is a special case of an ellipsoid
+# where each vertex has the same length. To fully specify an ellipsoid
+# in 3D, you need to supply three parameters. Two for the two spatial
+# dimensions of the image (via ``kernel_shape``), and one for the
+# intensity dimension (via ``intensity_vertex``).
+#
+# As mentioned above, a sphere is just a special ellipsoid, and hence
+# you can get the same behavior as ``rolling_ball`` if you set all
+# vertices to the respective values. In fact, ``rolling_ball``
+# internally calls ``rolling_spheroid`` in the way shown below.
+#
+# Note: The radius is equal to the length of a semi-vertex of a
+# sphere, which is *half* a full vertex. Hence, you need to multiply
+# the inputs by two if you want to get the same result as
+# ``rolling_ball``.
```

Comment:
On the radius topic: We are not rolling a ball/cap, but an ellipsoid. It has a slightly different shape than a ball/cap in particular around the borders of the kernel and for non-square kernels. I, therefore, switched to using terminology for an ellipsis (vertex) rather than using terminology used for a ball (radius).

Reply:
to be honest, I was today years old when I learned that ellipses have vertices. 😂 That's why the `intensity_vertex` nomenclature confused me so much! At any rate, from what I can tell [here](https://courses.lumenlearning.com/waymakercollegealgebra/chapter/equations-of-ellipses/), the vertex is only a point and we are after the major axis length? *And* based on [this page](https://mysite.du.edu/~jcalvert/math/ellipse.htm), an ellipse *does* have a radius of curvature, so perhaps `radius=` is ok?

I still need to get my brain straight as to why this is an ellipsoid and not the cap of a bigger sphere. Is it that you are approximating the cap as an ellipsoid? and is that approximation necessary?

Reply:
You opened pandoras box here @jni xD sorry for the long answer.

>  I was today years old when I learned that ellipses have vertices

I learned about them in secondary school, which was quite some time ago now x) Apparently, learning early is of no advantage here, since I am confusing terminology and am making things unnecessarily complicated.

> At any rate, from what I can tell here, the vertex is only a point and we are after the major axis length?

You are right that a vertex is just a point; somehow I remembered it as being the distance between the two points. 
Major and minor axes are only useful in 2D (nD would have many minor axes?); apparently the correct term is principal axis to refer to an arbitrary axis for [ellipsoids](https://en.wikipedia.org/wiki/Ellipsoid), and if we want the thing corresponding to radius it is principal semi-axis ... which I am certain almost nobody will make intuitive sense of.

The radius term from your second link refers to the [more general concept](https://en.wikipedia.org/wiki/Curvature) of the *(radius of) curvature of a surface at a certain point* (a good image for 1D is in the section "Osculating circle" in the link) and is a local property of a single point on some n-dimensional Riemannian manifold if you know what manifolds are. 

We can uniquely define curvature in the 1D case (ellipsis), but it will change depending on which point along the border we consider. For the ND case (we need at least 2D for the ellipsoid; it's a surface) we have to be satisfied with a curvature along a given direction. We could try choosing something that makes sense, e.g., the minimum curvature for all directions for a given point; in this case, the `radius` would be the radius of the smallest ball that we can fit to the surface of the ellipsoid. Unfortunately, this would again change based on where on the ellipsoid's surface we compute curvature, and at this point, I think we may have left the realm of "intuitive".

> I still need to get my brain straight as to why this is an ellipsoid and not the cap of a bigger sphere. Is it that you are approximating the cap as an ellipsoid? and is that approximation necessary?

I am not approximating the cap; it is because being the person writing the code and making some design decisions has it's little perks and I like making things unnecessarily complicated. 🤣 

The reason is actually related to the *radius of curvature* you brought up earlier and what that means for the level sets of the kernel. 

For the "cap of a ball" idea the kernel will have the same curvature everywhere on its surface (being a bit liberal and hand-wavy here ;) ), and it's level sets will be concentric circles around the center of the kernel. Slicing this has two pain points: (1) if the `kernel.shape` is significantly smaller than the radius then you filter with a rectangular kernel in the spatial dimensions, if it is of equal size as the radius you filter with a circular kernel, and in between you filter with a rectangular kernel with rounded edges; this is pretty weird behavior. (2) the level sets "fall off" the side of the kernel, which means that you will have uneven intensity around the border. This is not too big a problem per se, but it can introduce strange effects. For example, for `kernel.shape`s where one side is substantially larger than the other, you are computing a rolling ball filter along the long side of the kernel, but approximate a minimum filter along the short side. Again, not a huge issue, but potentially unintuitive.

For the ellipsoid idea, we are changing the curvature so that we "squash" the ball to always fit inside the kernel. The level sets never "fall off" the side, we use a kernel of same shape in all cases, and things just magically work out nicely. The downside is that you have to start thinking about ellipsoids an introduce complicated terminology . . .